pkg/kubelet/dockershim/docker_service.go

```diff
@@ -211,6 +212,7 @@ func NewDockerService(config *ClientConfig, podSandboxImage string, streamingCon
 		checkpointManager:         checkpointManager,
 		startLocalStreamingServer: startLocalStreamingServer,
 		networkReady:              make(map[string]bool),
+		containerCleanupInfos:     make(map[string]*containerCleanupInfo),
```

Comment:
how big can this map grow?  it looks like we ignore errors after cleanup and always remove from the map.  so its basically we try to clean up once, and after that oh well?

Reply:
> how big can this map grow?

Only as big as the number of containers which have been successfully created, and neither successfully started nor removed yet; _and_ which also have any clean up to be performed, ie that use GMSA cred specs.

> it looks like we ignore errors after cleanup and always remove from the map. so its basically we try to clean up once, and after that oh well?

Yes this is just best effort. These reg keys don't contain any secrets anyhow. Also, we do try to clean up again at each dockershim start.

Reply:
I think we may need periodic cleanup to ensure memory is not leaked permanently. 
I think this is okay for alpha, but would like this to be addressed by beta. Please add a comment and file an issue and/or update the KEP to reflect this (if @derekwaynecarr agrees with not doing this for alpha)

Reply:
@yujuhong : good call. Created https://github.com/kubernetes/kubernetes/issues/74843 and assigned it to myself.